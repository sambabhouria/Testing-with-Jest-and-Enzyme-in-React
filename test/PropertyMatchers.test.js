//Often there are fields in the object you want to snapshot which are generated (like IDs and Dates).
// If you try to snapshot these objects, they will force the snapshot to fail on every run:

// FOR FIRST TEST NOT FAIL BUT AFTER FAIL BECAOUSE TIME CHANGE USER : yarn test -u (for update snapshot)
it('will fail every time', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot();
});

//For these cases, Jest allows providing an asymmetric matcher for any property. 
//These matchers are checked before the snapshot is written or tested, and then saved to the snapshot file instead of the received value:

it('will check the matchers and pass', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
  });
});
