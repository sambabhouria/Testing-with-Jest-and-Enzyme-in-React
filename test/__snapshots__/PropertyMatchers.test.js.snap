// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`will fail every time 1`] = `
Object {
  "createdAt": 2021-03-11T14:40:16.692Z,
  "id": 6,
  "name": "LeBron James",
}
`;
